lib/fetch/formdata.js

```diff
@@ -157,8 +157,11 @@ class FormData {
     // replace the first such entry with entry and remove the others.
     const idx = this[kState].findIndex((entry) => entry.name === name)
     if (idx !== -1) {
-      this[kState][idx] = entry
-      this[kState] = this[kState].filter((entry) => entry.name !== name)
+      this[kState] = [].concat(
+        this[kState].slice(0, idx).filter((entry) => entry.name !== name),
+        [entry],
+        this[kState].slice(idx).filter((entry) => entry.name !== name),
+      )
```

Comment:
```js
const arr = this[kState]
arr[idx++] = entry
const len = arr.length
for (let n = idx; n < len; ++n) {
  if (arr[n].name === name) {
    continue
  }
  if (n !== idx) {
    arr[idx++] = arr[n]
  } else {
    idx++
  }
}
arr.length = idx

```

Reply:
```js
this[kState][idx++] = entry
this[kState] = [
  ...this[kState].slice(0, idx),
  ...this[kState].slice(idx).filter((entry) => entry.name !== name)
]
```

Reply:
Alternative ways of achieving the same thing with different complexity and performance.

Reply:
Oh yes. I forgot that we do not need to search `name` in the first part cuz it is the first match.